Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -3217,9 +3226,45 @@ void CGContextDrawRadialGradient(CGContextRef context,
                                  CGFloat endRadius,
                                  CGGradientDrawingOptions options) {
     NOISY_RETURN_IF_NULL(context);
+    NOISY_RETURN_IF_NULL(gradient);
     RETURN_IF(!context->ShouldDraw());
 
-    UNIMPLEMENTED();
+    // Note: We do not support cases where startCenter != endCenter
+    if (!CGPointEqualToPoint(startCenter, endCenter)) {
+        UNIMPLEMENTED_WITH_MSG("Do not support DrawRadialGradient for start/end centers that are not the same.");
+        return;
+    }
+
+    RETURN_IF(_CGGradientGetCount(gradient) == 0);
+
+    std::vector<D2D1_GRADIENT_STOP> gradientStops = __CGGradientToD2D1GradientStop(context, gradient, options, true);
+
+    ComPtr<ID2D1GradientStopCollection> gradientStopCollection;
+
+    ComPtr<ID2D1DeviceContext> deviceContext = context->DeviceContext();
+    FAIL_FAST_IF_FAILED(deviceContext->CreateGradientStopCollection(gradientStops.data(),
```

Comment:
Doesn't this need to go through the new error handling policies?  Thats is, we don't fail fast for enhanced error handling?